src/board_controller/BrainAlive/BrainAlive.cpp

```diff
@@ -0,0 +1,379 @@
+        }
+    }
+    else
+    {
+// https://github.com/OpenBluetoothToolbox/SimpleBLE/issues/26#issuecomment-955606799
+#ifdef __linux__
+        usleep (1000000);
+#endif
+    }
+
+    bool control_characteristics_found = false;
+
+    if (res == (int)BrainFlowExitCodes::STATUS_OK)
+    {
+        size_t services_count = simpleble_peripheral_services_count (brainalive_peripheral);
+        for (size_t i = 0; i < services_count; i++)
+        {
+            simpleble_service_t service;
+            if (simpleble_peripheral_services_get (brainalive_peripheral, i, &service) !=
+                SIMPLEBLE_SUCCESS)
+            {
+                safe_logger (spdlog::level::err, "failed to get service");
+                res = (int)BrainFlowExitCodes::BOARD_NOT_READY_ERROR;
+            }
+
+            safe_logger (spdlog::level::trace, "found servce {}", service.uuid.value);
+            for (size_t j = 0; j < service.characteristic_count; j++)
+            {
+                safe_logger (spdlog::level::trace, "found characteristic {}",
+                    service.characteristics[j].value);
+
+                if (strcmp (service.characteristics[j].value,
+                        "0000fe41-8e22-4541-9d4c-21edae82ed19") == 0) // Write Characteristics
+                {
+                    control_characteristics = std::pair<simpleble_uuid_t, simpleble_uuid_t> (
+                        service.uuid, service.characteristics[j]);
+                    control_characteristics_found = true;
+                    safe_logger (spdlog::level::info, "found control characteristic");
+                }
+                if (strcmp (service.characteristics[j].value,
+                        "0000fe42-8e22-4541-9d4c-21edae82ed19") == 0)  // Notification Characteristics
+                {
+                    if (simpleble_peripheral_notify (brainalive_peripheral, service.uuid,
+                            service.characteristics[j], ::read_notifications,
+                            (void *)this) == SIMPLEBLE_SUCCESS)
+                    {
+
+                        notified_characteristics.push_back (
+                            std::pair<simpleble_uuid_t, simpleble_uuid_t> (
+                                service.uuid, service.characteristics[j]));
+                    }
+                    else
+                    {
+                        safe_logger (spdlog::level::err, "Failed to notify for {} {}",
+                            service.uuid.value, service.characteristics[j].value);
+                        res = (int)BrainFlowExitCodes::GENERAL_ERROR;
+                    }
+                }
+            }
+        }
+    }
+    if ((res == (int)BrainFlowExitCodes::STATUS_OK) && (control_characteristics_found))
+    {
+        int buffer_size = board_descr["num_rows"].get<int> ();
+        current_buf.resize (12); // 12 eeg packages in single ble transaction
+        new_eeg_data.resize (4); // 4 eeg channels total
```

Comment:
it seems like your device sends data for all channels inside single ble transaction(looking at read_data method) it means that you don't need current_buf and new_eeg_data, they were used for Muse because there callbacks for different channels are in different BLE notifications and we had to store data between them